backend/Testing/LexCore/Services/UserServiceTest.cs

```diff
@@ -0,0 +1,71 @@
+using LexBoxApi.Auth;
+using LexBoxApi.Services;
+using LexBoxApi.Services.Email;
+using LexCore.Auth;
+using LexCore.Entities;
+using LexCore.ServiceInterfaces;
+using LexData;
+using Microsoft.AspNetCore.Http;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Caching.Memory;
+using Microsoft.Extensions.DependencyInjection;
+using Moq;
+using Npgsql;
+using Shouldly;
+using Testing.Fixtures;
+
+namespace Testing.LexCore.Services;
+
+[Collection(nameof(TestingServicesFixture))]
+public class UserServiceTest
+{
+    private readonly UserService _userService;
+
+    private readonly LexBoxDbContext _lexBoxDbContext;
+
+    public UserServiceTest(TestingServicesFixture testing)
+    {
+        var serviceProvider = testing.ConfigureServices(s =>
+        {
+            s.AddScoped<IEmailService>(_ => Mock.Of<IEmailService>());
+            s.AddScoped<UserService>();
+        });
+        _userService = serviceProvider.GetRequiredService<UserService>();
+        _lexBoxDbContext = serviceProvider.GetRequiredService<LexBoxDbContext>();
+    }
+
+    [Fact]
+    public async Task ManagerCanSeeAllUsersEvenInConfidentialProjects()
+    {
+        var manager = await _lexBoxDbContext.Users.Include(u => u.Organizations).Include(u => u.Projects).FirstOrDefaultAsync(user => user.Email == "manager@test.com");
+        manager.ShouldNotBeNull();
+        await using var privateProject = await TempProjectWithoutRepo.Create(_lexBoxDbContext, true, manager.Id);
+        privateProject.Project.ShouldNotBeNull();
+        var qaAdmin = await _lexBoxDbContext.Users.Include(u => u.Organizations).Include(u => u.Projects).FirstOrDefaultAsync(user => user.Email == "qa@test.com");
+        qaAdmin.ShouldNotBeNull();
+        privateProject.Project.Users.Add(new ProjectUsers() { UserId = qaAdmin.Id, Role = ProjectRole.Editor });
+        await _lexBoxDbContext.SaveChangesAsync();
+        var authUser = new LexAuthUser(manager);
+        var users = await _userService.UserQueryForTypeahead(authUser).ToArrayAsync();
+        users.ShouldNotBeEmpty();
+        users.ShouldContain(u => u.Id == qaAdmin.Id);
+    }
+
+    [Fact]
+    public async Task NonManagerCanNotSeeUsersInConfidentialProjects()
+    {
+        var manager = await _lexBoxDbContext.Users.Include(u => u.Organizations).Include(u => u.Projects).FirstOrDefaultAsync(user => user.Email == "manager@test.com");
+        manager.ShouldNotBeNull();
+        var editor = await _lexBoxDbContext.Users.Include(u => u.Organizations).Include(u => u.Projects).FirstOrDefaultAsync(user => user.Email == "editor@test.com");
+        editor.ShouldNotBeNull();
+        await using var privateProject = await TempProjectWithoutRepo.Create(_lexBoxDbContext, true, manager.Id);
```

Comment:
Yeah, that's not actually testing what I intended it to test until the editor is in the project. So what we want is some tests to verify the following:

* Admin can see everyone (which is a little pointless since they can already just query `users` and the frontend takes that into account, so maybe this test isn't necessary? It would require a code change anyway)
* Org members can see everyone in their org, and also anyone they share a public project with
* Project managers can see everyone in their projects, confidential or not, and also have the same rights as "normal" users in the projects they don't manage (can see public, can not see private)
* Non-managers can only see people they share public projects with (and org membership)

I think instead of using the users and projects from the seed data for this, I should have a test class constructor that sets up a specific set of projects and users, and a Dispose method on the test class that tears down all those projects and users. Then I can write the tests to the specific set of projects I created, without worrying about any changes I made to the seed data messing things up. (I once added some users to the sena-3 project and managed to make my tests start failing until I reset the data to the original seeded state. We don't want the tests to be that fragile).

Reply:
Commit af794179 tests more scenarios for projects, but doesn't actually include org membership in the test. So I'll need to add a few more users in a couple different orgs to really make these tests complete.